### ISSUE TITLE
Nvidium

### ISSUE THREAD

**Issue by Theoyeah** [Fabulously-Optimized/fabulously-optimized#752]
### CurseForge Mod Distribution

Unknown

### Modrinth link

https://modrinth.com/mod/nvidium

### Source/other link

https://github.com/MCRcortex/nvidium

### Mod file size

142.36 KB

### License

(any other license)

### What it does

The mod replaces the sodium rendering system and uses nvidia-only opengl extensions to increase framerates by a considerable amount.

With mesh shaders, a near fully gpu driven rendering pipeline is used, enabling very fast and performant geometry culling of terrain meaning your gpu can work much more efficiently.

### Why should it be in the modpack

It gives a lot of fps to RTX users (with that modpack + nvidium i get so much more fps compared to Bedrock Edition itself)

I have some screenshots to prove it (16 chunks & default settings for both editions but no anti-analysing in bedrock) 
![image](https://github.com/Fabulously-Optimized/fabulously-optimized/assets/97792861/8a9df1fe-4864-4540-9a16-0786ae49c5dc)
![image](https://github.com/Fabulously-Optimized/fabulously-optimized/assets/97792861/92feaf3c-32f4-4c97-9958-48ffceee9409)

### Why shouldn't it be in the modpack

Not available on Curseforge. (maybe it will one day ?)
The mod is on beta (Even if I tested it and i had no issues with the other mods included)
Only working for GTX 1600+ users and RTX users (note that the mod disables completely by itself if the graphics card isn't compatible)

**Comment by Madis0** [comment]
> > No, the mod is not functional on non-Nvidia systems, but the Nvidium will automatically disable itself. Your gameplay will not be affected.
>
> so this is the reason you and someone on [Discord] just suggested it for FO - it disables itself on non-Nvidia
I'm not sure it's a good argument yet, but it is a valid argument
what I would ask though is why is this even necessary
like, 200 fps without the mod is not enough? especially when it doesn't affect shaders in any way

**Comment by EquityX** [comment]
I tested it on the Hermitcraft Season 7 world & it smooths the framerate by a noticeable amount. This is 32 render distance with Nvidium.
![With Nvidium](https://github.com/Fabulously-Optimized/fabulously-optimized/assets/20719360/6694a4e4-743a-44b7-8b30-0885acb30ed2)

This is 32 render distance without Nvidium.
![Without Nvidium](https://github.com/Fabulously-Optimized/fabulously-optimized/assets/20719360/43396557-acbf-4af5-babc-f6b293a66508)

It also has a massive performance uplift with render distances past 32.
![Nvidium 64 Render Distance](https://github.com/Fabulously-Optimized/fabulously-optimized/assets/20719360/5c5049f6-91e8-41c5-8e81-04ed56fe8ab5)
64 Render distance & only a few fps lower than 32 Render distance.

The reason I would not add it yet is because it is still a beta mod & has some bugs. Like how teleporting with a command block will cause a crash. The dev has already responded & will push a fix in the future. Perhaps this could be added to a beta version of Fabuloulsy Optimized after the teleport bug is fixed to test its stability?

**Comment by Theoyeah** [comment]
Hey, seems like its been a long time since we discussed this. I would like to know if lately there have been any considerations adding it since it seems like today the mod seems to be a bit more free from critical bugs.

**Comment by TheBossMagnus** [comment]
> Hey, seems like its been a long time since we discussed this. I would like to know if lately there have been any considerations adding it since it seems like today the mod seems to be a bit more free from critical bugs.

Not really, but we have to note that the mod went from alpha to beta and the dev fixed quite a few bugs/memory leaks

**Comment by Madis0** [comment]
I'll request it in [non-CF mod list](https://support.curseforge.com/en/support/solutions/articles/9000197913-non-curseforge-mods) but I'll not consider it before Sodium 0.6 is out, due to its drastic changes.

**Comment by TheBossMagnus** [comment]
Little update on the mod: 
It should be out of beta soon (there is a release candidate in the discord, but some not-major issues are still in there). Could we make a test thread on discord after sodium 0.6 stable?

**Comment by TheBossMagnus** [comment]
nvidum is not yet compatible with sodium 0.6/minecraft 1.21.3, so for now it should stay on hold

**Comment by Atticuss26** [comment]
https://github.com/MCRcortex/nvidium/pull/282 There is currently a fork of nvidium with working 1.21.4 support.

**Comment by Madis0** [comment]
The mod hasn't been updated for a while and it "uses nvidia only opengl extensions", therefore it will not work and would not be needed with Vulkan (Minecraft 26.2+) anyway, closing.
